### PR TITLE
DrivAerML: 4L/256d+T_max=30 — gradient clipping max_norm=1.0

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Gradient clipping (max_norm=1.0) is a standard technique for stabilizing transformer training. The baseline model is still converging at the epoch cap (45 epochs), which could indicate unstable or noisy gradient updates during later epochs. Clipping gradients to a maximum L2 norm of 1.0 prevents large gradient spikes from destabilizing learned representations, potentially allowing the cosine LR schedule to drive the model further down the loss landscape during the final epochs. This is a low-risk intervention that is compatible with AdamW and cosine annealing.

## Instructions

Run the following command:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --max-grad-norm 1.0 \
  --wandb_group drivaerml-4L256d-training-tricks
```

Key changes from baseline:
- `--max-grad-norm 1.0` added (check train.py for the exact flag name — may be `--max-grad-norm`, `--clip-grad-norm`, `--gradient-clip-val`, `--grad-clip`, etc.)
- `--wandb_group drivaerml-4L256d-training-tricks` for grouping

All OOM-prevention flags are preserved: `--batch-size 1`, `--drivaerml-train-surface-points 50000`, `--drivaerml-eval-surface-points 50000`, `--max-train-batches 394`, `--max-eval-batches 200`, `--model-heads 4`.

Report val/loss and all surface MAE metrics (p_in, p_oodc, p_tan, p_re) and the primary metric `val_primary/surface_rel_l2_pct`. Compare against baseline of **12.70%**.

## Baseline

Current best: PR #2593 (shinji), W&B run: `3aaevlho`
- val_primary/surface_rel_l2_pct: **12.70%**
- Model: 4L/256d, T_max=30, AdamW lr=5e-4, no-EMA, Fourier features, 45 epochs (still converging)

Reproduce baseline:
```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2654 because ADVISOR: Closing — DrivAerML baseline shifted to 5.73% (4L/384d, PR #2602). All 4L/256d experiments are now obsolete — the best 4L/256d result was 11.97%. Width scaling is the dominant lever. Redirecting your GPU to 4L/384d experiments. You'll receive a new assignment shortly..
